File: GXpert/GXpert.Web/Modules/Analytics/AssignedExamTeachers/AssignedExamTeachersForm.cs
using Serenity.ComponentModel;
using System;

namespace GXpert.Analytics.Forms;

[FormScript("Analytics.AssignedExamTeachers")]
[BasedOnRow(typeof(AssignedExamTeachersRow), CheckNames = true)]
public class AssignedExamTeachersForm
{
    [HalfWidth]
    public int ExamId { get; set; }
    [HalfWidth]
    public int PlayListId { get; set; }
    [HalfWidth]
    public int TeacherId { get; set; }
}

File: GXpert/GXpert.Web/Modules/ServerTypes/Analytics/AssignedExamTeachersForm.ts
import { LookupEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface AssignedExamTeachersForm {
    ExamId: LookupEditor;
    PlayListId: LookupEditor;
    TeacherId: LookupEditor;
}

export class AssignedExamTeachersForm extends PrefixedContext {
    static readonly formKey = 'Analytics.AssignedExamTeachers';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!AssignedExamTeachersForm.init)  {
            AssignedExamTeachersForm.init = true;

            var w0 = LookupEditor;

            initFormType(AssignedExamTeachersForm, [
                'ExamId', w0,
                'PlayListId', w0,
                'TeacherId', w0
            ]);
        }
    }
}